### ISSUE TITLE
Fix rollback test to handle clusters with multiple Couchbase Servers

### ISSUE THREAD

**Issue by sethrosetter** [couchbaselabs/mobile-testkit#979]
The vbucket file may not be on the server we are targeting. 

- Reenable when fixed - https://github.com/couchbaselabs/mobile-testkit/blob/master/testsuites/syncgateway/functional/tests/test_rollback.py#L17-L24

**Comment by sethrosetter** [comment]
Sync with @tleyden or @adamcfraser before fixing this. 

The reason it works with one node server cluster and not 3. 
Testkit is not multinode aware when trying to delete the vbucket. 

The fix:
testkit needs to know how to resolve an Server node endpoint given a vBucket.

**Comment by sridevi-15** [comment]
Sync up with @adamcfraser for priority

**Comment by sridevi-15** [comment]
@adamcfraser , Could you let me know the expected behavior after multiple url support? 
Test still fails with multiple url support as test is  expecting sync-gateway fail to connect to server to get changes when server is restarted. It should retry atleast once to get changes through sync-gateway .


This is the code : 
https://github.com/couchbaselabs/mobile-testkit/blob/master/testsuites/syncgateway/functional/tests/test_rollback.py#L119-L137

It fails at Line 137.

**Comment by tleyden** [comment]
From discussion w / @sridevi-15 

- This comment from Seth https://github.com/couchbaselabs/mobile-testkit/issues/979#issuecomment-317901639 mentions that the test will need to be changed to delete the vbucket from the correct couchbase server node in order for the test to work with multiple couchbase server nodes.
- In addition to deleting the vbucket on the primary node, the vbucket may need to be deleted on the replica copy as well in order to simulate rollback behavior
- The `find -delete` command could be failing silently, needs more investigation to see what the process exit code when `find -delete` is called on non-existent file

**Comment by tleyden** [comment]
@sridevi-15 does this test pass or fail with a single couchbase server node?

**Comment by sridevi-15** [comment]
sg-config : 
```{
    "interface":":4984",
    "adminInterface": "0.0.0.0:4985",
    "maxIncomingConnections": 0,
    "maxCouchbaseConnections": 16,
    "maxFileDescriptors": 90000,
    "slowServerCallWarningThreshold": 500,
    "compressResponses": false,
    "log": ["CRUD+", "Cache+", "HTTP+", "Changes+", "DCP"],
    "cluster_config": {
        "server":"http://192.168.33.10,192.168.33.11,192.168.33.12:8091",
        "data_dir":".",
        "bucket":"data-bucket",
        "username":"data-bucket",
        "password": "password"
    },
    "databases":{
        "db":{
            "enable_shared_bucket_access": true,
            "server":"http://192.168.33.10,192.168.33.11,192.168.33.12:8091",
            "bucket":"data-bucket",
            "username":"data-bucket",
            "password": "password",
            "channel_index":{
                "num_shards":16,
                "server":"http://192.168.33.10,192.168.33.11,192.168.33.12:8091",
                "bucket":"index-bucket",
                "username": "index-bucket",
                "password": "password",
                "writer":false
            }
        }
    }
}
```

CBS version :5.0.0-3509
sg- version : 1.5.0-513
DI mode

**Comment by adamcfraser** [comment]
When running in di mode, the accel nodes are responsible for working the DCP feed.  You should see rollback notification in the accel logs as Warnings (even without DCP enabled), but adding the DCP log key to the accel config will provide even more rollback-related logging.

**Comment by sridevi-15** [comment]
[sync_gateway_error.zip](https://github.com/couchbaselabs/mobile-testkit/files/1279893/sync_gateway_error.zip)
[sg_accel_error.zip](https://github.com/couchbaselabs/mobile-testkit/files/1279894/sg_accel_error.zip)

I see DCP logs on sg-accel. Not sure if there is any logs related to rollback. Can you point me? I do not see any message related to rollback

**Comment by adamcfraser** [comment]
No - I see accel losing it's bucket connection, but I don't see any rollback processing.  This seems like it may have the same root cause as the lack of rollback notification in https://github.com/couchbase/sync_gateway/issues/2197, which should be addressed in 1.5.0-557.

**Comment by sridevi-15** [comment]
okay great, Should I try it on  1.5.0-557 and see if it works?
